### DISCUSSION TITLE
pdns-recursor returned his version on a dns txt request

### DISCUSSION THREAD

**Discussion by top-secrett** [PowerDNS/pdns#13582] [Q&A]
Hello everbody.
Today I've found that pdns-recursor answered his version.
Is it a bug or feature? How to prevent behavior like this?
```2023-12-07 05:04:46.597194082 pdns-recursor2 - CLIENT_QUERY 0 NOERROR 10.1.2.102 36472 10.1.2.131 53 6656 IPv4 UDP TXT VERSION.BIND 0.000000 0 - - - 104.234.254.138/32```
```2023-12-07 05:04:46.597791725 pdns-recursor2 - CLIENT_RESPONSE 0 NOERROR 10.1.2.102 36472 10.1.2.131 53 6656 IPv4 UDP TXT VERSION.BIND 0.000000 1 86400 PowerDNS Recursor 4.9.2 (built Nov  7 2023 08:30:24 by root@localhost) - 104.234.254.138/32```

**Comment by rgacogne** [comment]
https://docs.powerdns.com/recursor/settings.html#version-string

**Reply by top-secrett** [reply]
If I want to override this, I should add someting like this?
version-string="recursor"

**Reply by omoerbeek** [reply]
Yes. 

You might also want to check https://docs.powerdns.com/recursor/settings.html#server-id
